### PR TITLE
Atomic updates

### PR DESCRIPTION
**Pull request by serggl**
Add atomic updates to sunspot making possible to update individual properties of indexed document without having to post a bunch of unchanged fields

Inspired by https://wiki.apache.org/solr/Atomic_Updates.

### CODE REVIEW

**Comment by sergey-verevkin** [comment]
@serggl It is fantastic! I very like this implementation. Can't wait when it will be included to new release. Thank you a lot

**Comment by danmayer** [comment]
nice, looks good!

**Comment by lnavimanl** [comment]
@serggl 

Nice job! It's very useful thing!

**Comment by DmitryTsepelev** [comment]
:+1:

File: sunspot_rails/lib/sunspot/rails/stub_session_proxy.rb [hunk 1]
```diff
@@ -17,6 +17,12 @@ def index(*objects)
       def index!(*objects)
       end
 
+      def atomic_update(clazz, updates = {})
```
Comment (bragboy) [inline sunspot_rails/lib/sunspot/rails/stub_session_proxy.rb]: can you please add spec for these in `stub_session_proxy_spec.rb`?

File: sunspot/lib/sunspot/session_proxy/sharding_session_proxy.rb [hunk 2]
```diff
@@ -70,6 +70,20 @@ def index!(*objects)
         using_sharded_session(objects) { |session, group| session.index!(group) }
       end
 
+      #
+      # See Sunspot.atomic_update
+      #
+      def atomic_update(*objects)
```
Comment (bragboy) [inline sunspot/lib/sunspot/session_proxy/sharding_session_proxy.rb]: specs missing against `sharding_session_proxy_spec.rb`

File: sunspot/lib/sunspot/session.rb [hunk 3]
```diff
@@ -100,6 +100,22 @@ def index!(*objects)
     end
 
     #
+    # See Sunspot.atomic_update
+    #
+    def atomic_update(clazz, updates = {})
```
Comment (bragboy) [inline sunspot/lib/sunspot/session.rb]: specs please

File: sunspot_rails/lib/sunspot/rails/stub_session_proxy.rb (hunk 1 above)
Comment (serggl) [inline sunspot_rails/lib/sunspot/rails/stub_session_proxy.rb]: done

File: sunspot/lib/sunspot/session.rb (hunk 3 above)
Comment (serggl) [inline sunspot/lib/sunspot/session.rb]: I've added `integration\atomic_updates_spec.rb` is that what you were asking for?

File: sunspot/lib/sunspot/session_proxy/sharding_session_proxy.rb (hunk 2 above)
Comment (serggl) [inline sunspot/lib/sunspot/session_proxy/sharding_session_proxy.rb]: actually I marked `atomic_update` methods as `not_supported` for sharding proxies. I guess these classes were designed for pre-Solr4, where SolrCloud feature goes out-of-box

**Pull request merged**